Userland/Libraries/LibJS/Tests/builtins/Date/Date.prototype.toLocaleDateString.js

```diff
@@ -0,0 +1,49 @@
+describe("errors", () => {
+    test("called on non-Date object", () => {
+        expect(() => {
+            Date.prototype.toLocaleDateString();
+        }).toThrowWithMessage(TypeError, "Not an object of type Date");
+    });
+
+    test("called with value that cannot be converted to a number", () => {
+        expect(() => {
+            new Date(Symbol.hasInstance).toLocaleDateString();
+        }).toThrowWithMessage(TypeError, "Cannot convert symbol to number");
+
+        expect(() => {
+            new Date(1n).toLocaleDateString();
+        }).toThrowWithMessage(TypeError, "Cannot convert BigInt to number");
+    });
+
+    test("time value cannot be clipped", () => {
+        expect(() => {
+            new Date(-8.65e15).toLocaleDateString();
+        }).toThrowWithMessage(RangeError, "Time value must be between -8.64E15 and 8.64E15");
+    });
+
+    test("timeStyle may not be specified", () => {
+        expect(() => {
+            new Date().toLocaleDateString([], { timeStyle: "short" });
+        }).toThrowWithMessage(TypeError, "Option timeStyle cannot be set when also providing date");
+    });
+});
+
+describe("correct behavior", () => {
+    test("NaN", () => {
+        const d = new Date(NaN);
+        expect(d.toLocaleDateString()).toBe("Invalid Date");
+    });
+
+    const d0 = new Date(Date.UTC(2021, 11, 7, 17, 40, 50, 456));
+    const d1 = new Date(Date.UTC(1989, 0, 23, 7, 8, 9, 45));
+
+    test("defaults to date and time", () => {
+        expect(d0.toLocaleDateString()).toBe("12/7/2021");
+        expect(d1.toLocaleDateString()).toBe("1/23/1989");
+    });
+
+    test("dateStyle may be set", () => {
+        expect(d0.toLocaleDateString([], { dateStyle: "full" })).toBe("Tuesday, December 7, 2021");
+        expect(d1.toLocaleDateString([], { dateStyle: "full" })).toBe("Monday, January 23, 1989");
```

Comment:
No test for actually passing one or moore locales in the array?

Reply:
"I'll go back and add locales later", I thought to myself, then had ice cream and forgot :laughing: 
Fixed